driver/pgdriver/proto.go

```diff
@@ -121,7 +121,11 @@ func enableSSL(ctx context.Context, cn *Conn, tlsConf *tls.Config) error {
 		return errors.New("pgdriver: SSL is not enabled on the server")
 	}
 
-	cn.netConn = tls.Client(cn.netConn, tlsConf)
+	tlsCN := tls.Client(cn.netConn, tlsConf)
+	if err := tlsCN.HandshakeContext(ctx); err != nil {
```

Comment:
The handshake error would be hidden behind an `driver.ErrBadConn` otherwise. At least in this way the handshake error is returned. during the connection.

Reply:
Thanks for fixing this :+1: